test/NonDet.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE FlexibleContexts #-}
+module NonDet
+( tests
+) where
+
+import Control.Carrier.Error.Either
+import Control.Carrier.NonDet.Church
+import Control.Carrier.State.Strict
+import Prelude hiding (error)
+import Test.Tasty
+import Test.Tasty.HUnit
+
+tests :: TestTree
+tests = testGroup "NonDet"
```

Comment:
I really wanted to abstract these over the carrier so that we could test them properly against each carrier independently, but the [ridiculous contortions that required](https://github.com/fused-effects/fused-effects/pull/237/commits/08546e9e13734b13036fa984f4ecdadcc8fbc1de) were not worth the potential benefit.

New plan: can we express interpreting an effect as an effect? 🤔